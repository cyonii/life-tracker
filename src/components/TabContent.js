import Tab from 'react-bootstrap/Tab';
import AddRecord from './panes/AddRecord';
import Records from './panes/Records';
import ProgressReport from './panes/ProgressReport';
import More from './panes/More';

const TabContent = () => {
  const pane = (eventKey, title, component = null) => (
    <Tab.Pane eventKey={eventKey}>
      <h3 className="h4 bg-primary text-center text-white py-2">{title}</h3>
      {component}
    </Tab.Pane>
  );

  return (
    <Tab.Content>
      {pane('add-record', 'Add record', <AddRecord />)}
      {pane('records', 'Track.it', <Records />)}
      {pane('progress-report', 'Progress report', <ProgressReport />)}
      {pane('more', 'More', <More />)}
    </Tab.Content>
  );
};

export default TabContent;
