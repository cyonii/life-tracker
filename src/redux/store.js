import { createStore, combineReducers } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import records from './reducers/records';

export default createStore(combineReducers({ records }), composeWithDevTools());
