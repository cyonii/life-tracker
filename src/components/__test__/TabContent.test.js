import { screen, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import TabContent from '../TabContent';

describe('TabContent', () => {
  it('should render correctly', () => {
    render(
      <Provider store={store}>
        <TabContent />
      </Provider>,
    );
    expect(screen.getByTestId('tab-content')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <TabContent />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
