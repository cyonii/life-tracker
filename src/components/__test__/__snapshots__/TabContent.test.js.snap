// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabContent should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="tab-content"
    data-testid="tab-content"
  >
    <div
      class="tab-pane"
    >
      <h3
        class="h4 bg-primary text-center text-white py-2"
      >
        Add record
      </h3>
      <div
        class="container"
      />
    </div>
    <div
      class="tab-pane"
    >
      <h3
        class="h4 bg-primary text-center text-white py-2"
      >
        Track.it
      </h3>
      <div
        class="container"
      >
        <div
          class="p-0 list-group list-group-flush"
        />
      </div>
    </div>
    <div
      class="tab-pane"
    >
      <h3
        class="h4 bg-primary text-center text-white py-2"
      >
        Progress report
      </h3>
      <div
        class="container"
      >
        <h2
          class="text-muted"
        >
          Your Progress so far
        </h2>
      </div>
    </div>
    <div
      class="tab-pane"
    >
      <h3
        class="h4 bg-primary text-center text-white py-2"
      >
        More
      </h3>
      <div>
        <div
          class="list-group list-group-flush"
        >
          <div
            class="list-group-item"
          >
            <button
              class="w-100 rounded-0 btn btn-light"
              disabled=""
              type="button"
            >
              Profile
            </button>
            <button
              class="w-100 rounded-0 btn btn-danger"
              type="button"
            >
              Logout
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
